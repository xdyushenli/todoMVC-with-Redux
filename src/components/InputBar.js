import React from 'react';
import '../index.css';

export default class InputBar extends React.Component{
    constructor(props) {
        super(props);

        this.state = {inputText: ''};
    }

    handleSelectAll() {
        this.props.selectAll();
    }

    handleInputText(e) {
        this.setState({inputText: e.target.value});
    }

    handleAddItem(e) {
        if(e.key === "Enter") {
            this.props.addItem(this.state.inputText);
            this.setState({inputText: ""});
        }
    }

    render() {
        return ( 
            <div id='input-bar'>
                <div    className='select-all-btn'
                        onClick={() => this.handleSelectAll()}
                ></div>
                <input  type="text" 
                        className='input-box' 
                        placeholder='What needs to be done?'
                        value={this.state.inputText} 
                        onChange={(e) => this.handleInputText(e)}  
                        onKeyPress={(e) => this.handleAddItem(e)}     
                />
            </div>
        );
    }
}