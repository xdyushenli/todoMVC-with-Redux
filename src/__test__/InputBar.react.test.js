import React from 'react';
import InputBar from '../components/InputBar';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

it('render test', () => {
    const tree = renderer.create(
        <InputBar></InputBar>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('state should be {inputText: "some text"}', () => {
    const component = shallow(<InputBar></InputBar>);
    component.find('input').simulate('change', {target: {value: 'some text'}});
    
    expect(component.state()).toEqual({inputText: 'some text'});
})