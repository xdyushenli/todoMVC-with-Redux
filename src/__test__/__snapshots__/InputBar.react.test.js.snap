// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
<div
  id="input-bar"
>
  <div
    className="select-all-btn"
    onClick={[Function]}
  />
  <input
    className="input-box"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
</div>
`;
